extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/hll/sql/HllSketchSqlAggregatorTest.java

```diff
@@ -508,6 +524,33 @@ public void testApproxCountDistinctHllSketchIsRounded()
     );
   }
 
+  @Test
+  public void testApproxCountDistinctOnUnsupportedComplexColumn()
+  {
+    try {
+      testQuery("SELECT COUNT(distinct double_first_added) FROM druid.wikipedia_first_last", ImmutableList.of(), ImmutableList.of());
+      Assert.fail("query planning should fail");
+    }
+    catch (DruidException e) {
+      Assert.assertEquals("Query could not be planned. A possible reason is [Using APPROX_COUNT_DISTINCT() or enabling approximation "
```

Comment:
Why does the SQL layer not catch it? Is it because we are bypassing the type check when using `COUNT(DiSTINCT ...)`?

Reply:
Yep